typescript/packages/subsurface-viewer/src/layers/wells/wellsLayer.ts

```diff
@@ -251,13 +251,19 @@ export default class WellsLayer extends CompositeLayer<WellsLayerProps> {
 
             removeDuplicates(data);
 
-            const refine = this.props.refine;
-            data = refine
-                ? splineRefine(data) // smooth well paths.
-                : data;
-
             const coarseData = coarsenWells(data);
 
+            const doRefine =
+                typeof this.props.refine === "number"
+                    ? (this.props.refine as number) > 1
+                    : (this.props.refine as boolean);
+
+            const noRefineSteps =
```

Comment:
noRefineSteps ? no prefix is misleading. You can replace by nbRefineSteps or imply refineSteps ;)